### ISSUE TITLE
Failing to add new filters on Macbook M1

### ISSUE THREAD

**Issue by torluca** [dfm/python-fsps#176]
Hello,

I installed FSPS via pip on a Macbook Pro M1 and the installation went smoothly. I then tried to add new filters following the instructions on the manual. The filters were not recognized, therefore I uninstalled fsps and installed it with the FLAGS highlighted on the webpage

pip uninstall fsps
FFLAGS="-DMIST=0 -DPADOVA=1 -DMILES=1 -DBASEL=0" python -m pip install fsps --no-binary fsps

This approach works on a regular intel-based mac, but not on the M1 macs. The installation is successful, but when I try to import fsps, the following error message appears:

ImportError: dlopen(/Users/Luca.Tortorelli/opt/anaconda3/envs/astro_env/lib/python3.9/site-packages/fsps/_fsps.cpython-39-darwin.so, 0x0002): tried: '/Users/Luca.Tortorelli/opt/anaconda3/envs/astro_env/lib/python3.9/site-packages/fsps/_fsps.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/_fsps.cpython-39-darwin.so' (no such file), '/usr/lib/_fsps.cpython-39-darwin.so' (no such file)

Do you know how to solve this issue?

Thanks.

**Comment by torluca** [comment]
Btw there is a mismatch between the number of filters hard-coded in the Fortran version of FSPS and in its Python version. In the latter, only 143 filters are present, while in the former they are 159. Basically the LSST, Euclid and Roman filters cannot be loaded.

**Comment by dfm** [comment]
I can help with the compiling issue and perhaps @bd-j can make some comments about the filters.

I can also confirm that the above command works just fine on my M1 mac, so this not a generic problem with all M1s. That being said, I'm sure we can track down the issue.

The way the error message is worded suggests to me that you may actually be running a virtualized Python. What is the full output that you get when you run the following three commands in your terminal:

```bash
arch
conda info
python -c "import platform; print(platform.uname())"
```

**Comment by torluca** [comment]
Thanks a lot for your fast reply!

The output of the three commands are:

(astro_env) usm-nb-lgnmbp08:~ Luca.Tortorelli$ arch
arm64
(astro_env) usm-nb-lgnmbp08:~ Luca.Tortorelli$ conda info

     active environment : astro_env
    active env location : /Users/Luca.Tortorelli/opt/anaconda3/envs/astro_env
            shell level : 1
       user config file : /Users/Luca.Tortorelli/.condarc
 populated config files : /Users/Luca.Tortorelli/.condarc
          conda version : 4.11.0
    conda-build version : 3.21.5
         python version : 3.9.7.final.0
       virtual packages : __osx=10.16=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /Users/Luca.Tortorelli/opt/anaconda3  (writable)
      conda av data dir : /Users/Luca.Tortorelli/opt/anaconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/Luca.Tortorelli/opt/anaconda3/pkgs
                          /Users/Luca.Tortorelli/.conda/pkgs
       envs directories : /Users/Luca.Tortorelli/opt/anaconda3/envs
                          /Users/Luca.Tortorelli/.conda/envs
               platform : osx-64
             user-agent : conda/4.11.0 requests/2.26.0 CPython/3.9.7 Darwin/21.3.0 OSX/10.16
                UID:GID : 503:20
             netrc file : None
           offline mode : False

(astro_env) usm-nb-lgnmbp08:~ Luca.Tortorelli$ python -c "import platform; print(platform.uname())"
uname_result(system='Darwin', node='usm-nb-lgnmbp08.local', release='21.3.0', version='Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000', machine='x86_64')

**Comment by dfm** [comment]
Ok great - so as expected your Python thinks that it's running on an Intel machine, not an M1! (ref: the `machine='x86_64'` in the last line)

But I guess it's somehow not communicating that to the compilers when actually compiling FSPS. This is probably something that we can track down, but I'd probably recommend switching to the `miniforge` Python that comes with a proper `arm64` environment so that it doesn't have to run virtualized: https://github.com/conda-forge/miniforge#miniforge3

**Comment by torluca** [comment]
Ok, I see, I'll give it a try and come back to you. Thanks again.

**Comment by bd-j** [comment]
re the filters, those were added to FSPS just after the last python-FSPS release.  Installing the dev version of python-fsps would get them, but I should probably draft a new release as there were a couple other small updates.

**Comment by bd-j** [comment]
(see also https://github.com/bd-j/sedpy for a package to do filter projections that is a bit more flexible than the giant allfilters.dat file in fsps)

**Comment by torluca** [comment]
Installing conda from miniforge solved the issue. Thanks very much @dfm !

@bd-j I downloaded the dev version and the additional filters were there, so it's just a problem of the package hosted on Pypi. Thanks for pointing sedpy to me btw. 

Just for the record, the filters I added are from the Physics of the Accelerated Universe survey, which is a narrow-band photometric survey. If you plan to include additional filters, those can be found at https://pausurvey.org/paucam/filters/
